drivers/npmx/npmx_shell.c

```diff
@@ -311,34 +311,402 @@ static npmx_instance_t *npmx_instance_get(const struct shell *shell)
+
+	*resistance = (uint32_t)resistance_float;
+
+	return true;
+}
+
+static int cmd_adc_meas_vbat_get(const struct shell *shell, size_t argc, char **argv)
+{
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	npmx_adc_t *adc_instance = adc_instance_get(shell);
+	if (adc_instance == NULL) {
+		return 0;
+	}
+
+	npmx_error_t err_code = npmx_adc_task_trigger(adc_instance, NPMX_ADC_TASK_SINGLE_SHOT_VBAT);
+	if (!check_error_code(shell, err_code)) {
+		shell_error(shell, "Error: unable to trigger the measurement.");
+		return 0;
+	}
+
+	int32_t voltage_mv;
+	bool meas_get = false;
+	while (!meas_get) {
+		err_code = npmx_adc_meas_check(adc_instance, NPMX_ADC_MEAS_VBAT, &meas_get);
+		if (!check_error_code(shell, err_code)) {
+			print_get_error(shell, "measurement status");
+			return 0;
+		}
+	}
+
+	err_code = npmx_adc_meas_get(adc_instance, NPMX_ADC_MEAS_VBAT, &voltage_mv);
+	if (!check_error_code(shell, err_code)) {
+		print_get_error(shell, "measurement");
+		return 0;
+	}
+
+	print_value(shell, voltage_mv, UNIT_TYPE_MILLIVOLT);
+	return 0;
+}
+
+static bool adc_ntc_temperatures_recalculate(const struct shell *shell,
+					     temp_recalculate_data_t const *data)
+{
+	int16_t temperature;
+	uint32_t resistance;
+
+	npmx_error_t err_code = data->get_resistance_func(data->charger_instance, &resistance);
+	if (!check_error_code(shell, err_code)) {
+		print_get_error(shell, "old resistance");
+		return false;
+	}
+	if (!ntc_resistance_to_temperature(shell, data->ntc_config_old, resistance, &temperature)) {
+		print_convert_error(shell, "old resistance", "temperature");
+		return false;
+	}
+	if (!ntc_temperature_to_resistance(shell, data->ntc_config_new, temperature, &resistance)) {
+		print_convert_error(shell, "temperature", "new resistance");
+		return false;
+	}
+	err_code = data->set_resistance_func(data->charger_instance, resistance);
+
+	if (!check_error_code(shell, err_code)) {
+		print_set_error(shell, "new resistance");
+		return false;
+	}
+
+	return true;
+}
+
+static int adc_ntc_set(const struct shell *shell, size_t argc, char **argv,
+		       adc_ntc_config_param_t config_type)
+{
+	char *config_name;
+	switch (config_type) {
+	case ADC_NTC_CONFIG_PARAM_TYPE:
+		config_name = "NTC type";
+		break;
+	case ADC_NTC_CONFIG_PARAM_BETA:
+		config_name = "NTC beta";
+		break;
+	}
+
+	args_info_t args_info = { .expected_args = 1,
+				  .arg = {
+					  [0] = { SHELL_ARG_TYPE_UINT32_VALUE, config_name },
+				  } };
+	if (!arguments_check(shell, argc, argv, &args_info)) {
+		return 0;
+	}
+
+	uint32_t config_value = args_info.arg[0].result.uvalue;
+	if ((config_type == ADC_NTC_CONFIG_PARAM_BETA) && (config_value == 0)) {
+		shell_error(shell, "Error: beta cannot be equal to zero.");
+		return 0;
+	}
+
+	npmx_charger_t *charger_instance = charger_instance_get(shell);
+	if (charger_instance == NULL) {
+		return 0;
+	}
+
+	if (!charger_disabled_check(shell, charger_instance, "NTC config")) {
+		return 0;
+	}
```

Comment:
Those if's could be merged.

Reply:
I tried to evaluate all `ifs` and `returns` to be a separate case.

Reply:
Why?